### PR TITLE
Proper certificate for meet.jitsi internal domain

### LINKED ISSUE #240: Recording fails
Hi everyone,
I tried to enable jibri and encountered some problems. First I got the following error message:

```
org.jitsi.jibri.service.impl.FileRecordingJibriService.finalize() Failed to run finalize script: java.io.IOException: Cannot run program "/config/finalize.sh": error=2, No such file or directory
```

But this was easy to fix. I just created a file in my config folder, but now I get the following:

```
org.jitsi.jibri.selenium.JibriSelenium.run() An error occurred while joining the call: org.openqa.selenium.WebDriverException: <unknown>: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
```

I think the problem is that `Window` is different from `window`, but I don't know where to find the code. Can you help me?

<details>
<summary>Complete log</summary>
```
2020-03-20 09:15:16.071 FINE: [35] org.jitsi.xmpp.mucclient.MucClient.log() Received an IQ with type set: IQ Stanza (jibri http://jitsi.org/protocol/jibri) [to=jibri@auth.meet.jitsi/edAXYQ94,from=jibribrewery@internal-muc.meet.jitsi/focus,id=amlicmlAYXV0aC5tZWV0LmppdHNpL2VkQVhZUTk0ADBCVGx1LTI1NABVsWt3X1n/o6vD6m2Rlbe7,type=set,]
2020-03-20 09:15:16.073 INFO: [35] org.jitsi.jibri.api.xmpp.XmppApi.handleJibriIq() Received JibriIq <iq to='jibri@auth.meet.jitsi/edAXYQ94' from='jibribrewery@internal-muc.meet.jitsi/focus' id='amlicmlAYXV0aC5tZWV0LmppdHNpL2VkQVhZUTk0ADBCVGx1LTI1NABVsWt3X1n/o6vD6m2Rlbe7' type='set'><jibri xmlns='http://jitsi.org/protocol/jibri' action='start' recording_mode='file' room='asdfxxssexfadsf@muc.meet.jitsi' session_id='fbzkuvzxovfdfkxh' app_data='{"file_recording_metadata":{"share":true}}'/></iq> from environment [MucClient id=xmpp.meet.jitsi hostname=xmpp.meet.jitsi]
2020-03-20 09:15:16.075 INFO: [35] org.jitsi.jibri.api.xmpp.XmppApi.handleStartJibriIq() Received start request
2020-03-20 09:15:16.078 INFO: [35] org.jitsi.jibri.api.xmpp.XmppApi.handleStartJibriIq() Sending 'pending' response to start IQ
2020-03-20 09:15:16.078 INFO: [38] org.jitsi.jibri.api.xmpp.XmppApi.run() Starting service
2020-03-20 09:15:16.107 INFO: [38] org.jitsi.jibri.api.xmpp.XmppApi.handleStartService() Parsed call url info: CallUrlInfo(baseUrl=https://meet.jitsi, callName=asdfxxssexfadsf, urlParams=[])
2020-03-20 09:15:16.108 INFO: [38] org.jitsi.jibri.JibriManager.startFileRecording() Starting a file recording with params: FileRecordingRequestParams(callParams=CallParams(callUrlInfo=CallUrlInfo(baseUrl=https://meet.jitsi, callName=asdfxxssexfadsf, urlParams=[])), sessionId=fbzkuvzxovfdfkxh, callLoginParams=XmppCredentials(domain=recorder.meet.jitsi, username=recorder, password=passw0rd)) finalize script path: /config/finalize.sh and recordings directory: /config/recordings
2020-03-20 09:15:16.764 INFO: [38] org.openqa.selenium.remote.ProtocolHandshake.createSession() Detected dialect: OSS
2020-03-20 09:15:16.779 INFO: [38] org.jitsi.jibri.selenium.JibriSelenium.<init>() Starting empty call check with a timeout of PT30S
2020-03-20 09:15:16.790 FINE: [38] org.jitsi.jibri.capture.ffmpeg.FfmpegCapturer.<init>() Detected os as OS: LINUX
2020-03-20 09:15:16.794 INFO: [38] org.jitsi.jibri.service.impl.FileRecordingJibriService.<init>() Writing recording to /config/recordings/fbzkuvzxovfdfkxh
2020-03-20 09:15:16.796 FINE: [38] org.jitsi.jibri.statsd.JibriStatsDClient.incrementCounter() Incrementing statsd counter: start:recording
2020-03-20 09:15:16.797 INFO: [38] org.jitsi.jibri.status.JibriStatusManager.log() Busy status has changed: IDLE -> BUSY
2020-03-20 09:15:16.797 INFO: [38] org.jitsi.jibri.api.xmpp.XmppApi.updatePresence() Jibri reports its status is now JibriStatus(busyStatus=BUSY, health=OverallHealth(healthStatus=HEALTHY, details={})), publishing presence to connections
2020-03-20 09:15:16.798 INFO: [38] org.jitsi.xmpp.mucclient.MucClientManager.log() Setting a presence extension: org.jitsi.xmpp.extensions.jibri.JibriStatusPacketExt@504ca55b
2020-03-20 09:15:16.798 FINE: [38] org.jitsi.xmpp.mucclient.MucClientManager.log() Replacing presence extension: org.jitsi.xmpp.extensions.jibri.JibriStatusPacketExt@5aa6202e
2020-03-20 09:15:16.800 FINE: [32] org.jitsi.xmpp.extensions.DefaultPacketExtensionProvider.parse() Could not add a provider for element busy-status from namespace http://jitsi.org/protocol/jibri
2020-03-20 09:15:16.801 FINE: [32] org.jitsi.xmpp.extensions.DefaultPacketExtensionProvider.parse() Could not add a provider for element health-status from namespace http://jitsi.org/protocol/health
2020-03-20 09:15:16.928 SEVERE: [48] org.jitsi.jibri.selenium.JibriSelenium.run() An error occurred while joining the call: org.openqa.selenium.WebDriverException: <unknown>: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
  (Session info: chrome=78.0.3904.108)
  (Driver info: chromedriver=78.0.3904.105 (60e2d8774a8151efa6a00b1f358371b1e0e07ee2-refs/branch-heads/3904@{#877}),platform=Linux 4.15.0-91-generic x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 0 milliseconds
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: '6648c4582aae', ip: '172.18.0.4', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-91-generic', java.version: '1.8.0_222'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, acceptSslCerts: false, applicationCacheEnabled: false, browserConnectionEnabled: false, browserName: chrome, chrome: {chromedriverVersion: 78.0.3904.105 (60e2d8774a81..., userDataDir: /tmp/.com.google.Chrome.cJTEvn}, cssSelectorsEnabled: true, databaseEnabled: false, goog:chromeOptions: {debuggerAddress: localhost:41075}, handlesAlerts: true, hasTouchScreen: false, javascriptEnabled: true, locationContextEnabled: true, mobileEmulationEnabled: false, nativeEvents: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, proxy: Proxy(), rotatable: false, setWindowRect: true, strictFileInteractability: false, takesHeapSnapshot: true, takesScreenshot: true, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unexpectedAlertBehaviour: ignore, unhandledPromptBehavior: ignore, version: 78.0.3904.108, webStorageEnabled: true}
Session ID: c7d783873cf12c9cf21c44af1ca43a71 with stack: 
sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
java.lang.reflect.Constructor.newInstance(Constructor.java:423)
org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:214)
org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:166)
org.openqa.selenium.remote.http.JsonHttpResponseCodec.reconstructValue(JsonHttpResponseCodec.java:40)
org.openqa.selenium.remote.http.AbstractHttpResponseCodec.decode(AbstractHttpResponseCodec.java:80)
org.openqa.selenium.remote.http.AbstractHttpResponseCodec.decode(AbstractHttpResponseCodec.java:44)
org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:158)
org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:543)
org.openqa.selenium.remote.RemoteWebDriver.executeScript(RemoteWebDriver.java:480)
org.jitsi.jibri.selenium.JibriSelenium.setLocalStorageValues(JibriSelenium.kt:175)
org.jitsi.jibri.selenium.JibriSelenium.access$setLocalStorageValues(JibriSelenium.kt:112)
org.jitsi.jibri.selenium.JibriSelenium$joinCall$1.run(JibriSelenium.kt:255)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)

2020-03-20 09:15:16.931 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.onSeleniumStateChange() Transitioning from state Starting up to Error: SESSION Failed to join call
2020-03-20 09:15:16.931 INFO: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.onServiceStateChange() File recording service transitioning from state Starting up to Error: SESSION Failed to join call
2020-03-20 09:15:16.932 INFO: [48] org.jitsi.jibri.api.xmpp.XmppApi.invoke() Current service had an error, sending error iq <iq to='jibribrewery@internal-muc.meet.jitsi/focus' id='eI5pF-27' type='set'><jibri xmlns='http://jitsi.org/protocol/jibri' status='off' failure_reason='error'/></iq>
2020-03-20 09:15:16.932 FINE: [48] org.jitsi.jibri.statsd.JibriStatsDClient.incrementCounter() Incrementing statsd counter: stop:recording
2020-03-20 09:15:16.933 INFO: [48] org.jitsi.jibri.JibriManager.stopService() Stopping the current service
2020-03-20 09:15:16.933 INFO: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.stop() Stopping capturer
2020-03-20 09:15:16.934 INFO: [48] org.jitsi.jibri.util.JibriSubprocess.ffmpeg.stop() Stopping ffmpeg process
2020-03-20 09:15:16.934 INFO: [48] org.jitsi.jibri.util.JibriSubprocess.ffmpeg.stop() ffmpeg exited with value null
2020-03-20 09:15:16.934 INFO: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.stop() Quitting selenium
2020-03-20 09:15:16.943 INFO: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.stop() Participants in this recording: []
2020-03-20 09:15:16.987 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.leaveCallAndQuitBrowser() Got 0 log entries for type browser
2020-03-20 09:15:17.001 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.leaveCallAndQuitBrowser() Got 110 log entries for type driver
2020-03-20 09:15:17.026 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.leaveCallAndQuitBrowser() Got 0 log entries for type client
2020-03-20 09:15:17.027 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.leaveCallAndQuitBrowser() Leaving web call
2020-03-20 09:15:17.045 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.leaveCallAndQuitBrowser() Quitting chrome driver
2020-03-20 09:15:17.116 INFO: [48] org.jitsi.jibri.selenium.JibriSelenium.leaveCallAndQuitBrowser() Chrome driver quit
2020-03-20 09:15:17.116 INFO: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.stop() Finalizing the recording
2020-03-20 09:15:27.125 SEVERE: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.finalize() Timed out waiting for process logger task to complete
2020-03-20 09:15:27.126 INFO: [48] org.jitsi.jibri.service.impl.FileRecordingJibriService.finalize() Recording finalize script finished with exit value 0
2020-03-20 09:15:27.127 INFO: [48] org.jitsi.jibri.status.JibriStatusManager.log() Busy status has changed: BUSY -> IDLE
2020-03-20 09:15:27.128 INFO: [48] org.jitsi.jibri.api.xmpp.XmppApi.updatePresence() Jibri reports its status is now JibriStatus(busyStatus=IDLE, health=OverallHealth(healthStatus=HEALTHY, details={})), publishing presence to connections
2020-03-20 09:15:27.128 INFO: [48] org.jitsi.xmpp.mucclient.MucClientManager.log() Setting a presence extension: org.jitsi.xmpp.extensions.jibri.JibriStatusPacketExt@754e0f5b
2020-03-20 09:15:27.129 FINE: [48] org.jitsi.xmpp.mucclient.MucClientManager.log() Replacing presence extension: org.jitsi.xmpp.extensions.jibri.JibriStatusPacketExt@504ca55b
2020-03-20 09:15:27.132 FINE: [32] org.jitsi.xmpp.extensions.DefaultPacketExtensionProvider.parse() Could not add a provider for element busy-status from namespace http://jitsi.org/protocol/jibri
2020-03-20 09:15:27.132 FINE: [32] org.jitsi.xmpp.extensions.DefaultPacketExtensionProvider.parse() Could not add a provider for element health-status from namespace http://jitsi.org/protocol/health
```
</details>

### PR DESCRIPTION
**Pull request by teowoz**
- web: generate self-signed certificate with subjectAltNames
- jibri: insert it into chrome chain

This should fix #240, namely this error:

`org.jitsi.jibri.selenium.JibriSelenium.run() An error occurred while joining the call: org.openqa.selenium.WebDriverException: <unknown>: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.`

### CODE REVIEW

**Comment by ricardona** [comment]
@teowoz I tried [your instructions](https://github.com/jitsi/docker-jitsi-meet/issues/240#issuecomment-611633063) but I got:

```
jibri_1    | 2020-04-10 00:25:15.393 FINE: [61] org.jitsi.jibri.selenium.pageobjects.CallPage.visit()Visiting url https://meet.jitsi/pimpollo#config.iAmRecorder=true&config.externalConnectUrl=null&config.startWithAudioMuted=true&config.startWithVideoMuted=true&interfaceConfig.APP_NAME="Jibri"&config.analytics.disabled=true&config.p2p.enabled=false

jibri_1    | 2020-04-10 00:25:46.112 SEVERE: [61] org.jitsi.jibri.selenium.pageobjects.CallPage.visit() Timed out waiting for call page to load
```

**Comment by teowoz** [comment]
@ricardona 

Did it successfully add the certificate?
It should print:
> Updating certificates in /etc/ssl/certs...
> 1 added, 0 removed; done.

during start.
Before checking it, remove (stopping is not enough) the jibri container.

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/with-contenv bash
+
+while ! openssl s_client -showcerts -connect meet.jitsi:443 < /dev/null | openssl x509 -outform PEM > /usr/local/share/ca-certificates/meet.crt; do sleep 1; done
```
Comment (saghul) [inline jibri/rootfs/etc/cont-init.d/11-get-https-cert:3]: This won't work if the port is not 443. I know there are other problems in that case, but we should use the env var here.

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+update-ca-certificates
+# the screenshot is just to launch chrome and exit, to make it create ~/.pki
+su jibri -s /bin/sh -c '/usr/bin/google-chrome --headless --screenshot=/tmp/s.png https://meet.jitsi/ && certutil -d sql:/home/jibri/.pki/nssdb -A -t TC -n meet -i /usr/local/share/ca-certificates/meet.crt'
```
Comment (saghul) [inline jibri/rootfs/etc/cont-init.d/11-get-https-cert:6]: Can you explain this sorcery? I love it!

File: web/rootfs/defaults/ssl.conf [hunk 3]
```diff
@@ -22,6 +22,6 @@ ssl_ecdh_curve secp384r1;
 ssl_ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDH+CHACHA20:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!3DES:!MD5:!PSK;
 
 # headers
-add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
+#add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
```
Comment (saghul) [inline web/rootfs/defaults/ssl.conf:25]: Why is this change needed?

File: web/rootfs/etc/cont-init.d/10-config [hunk 4]
```diff
@@ -43,8 +43,11 @@ if [[ $DISABLE_HTTPS -ne 1 ]]; then
             echo "using keys found in /config/keys"
         else
             echo "generating self-signed keys in /config/keys, you can replace these with your own keys if required"
-            SUBJECT="/C=US/ST=TX/L=Austin/O=jitsi.org/OU=Jitsi Server/CN=*"
-            openssl req -new -x509 -days 3650 -nodes -out /config/keys/cert.crt -keyout /config/keys/cert.key -subj "$SUBJECT"
+            openssl req -newkey rsa:2048 -new -x509 -days 9131 -nodes -sha256 \
```
Comment (saghul) [inline web/rootfs/etc/cont-init.d/10-config:46]: do we really need more than 10 years?

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert (hunk 1 above)
Comment (albe) [inline jibri/rootfs/etc/cont-init.d/11-get-https-cert:3]: Tried this PR, but I get errors (certificate in the web container is created correctly):

```
openssl s_client -showcerts -connect meet.jitsi:443 < /dev/null
139985578098752:error:0200206F:system library:connect:Connection refused:../crypto/bio/b_sock2.c:108:
139985578098752:error:2008A067:BIO routines:BIO_connect:connect error:../crypto/bio/b_sock2.c:109:
connect:errno=111
```

I even adjusted the port to my local setup (8443), where I overrode the web port because the docker containers run behind a proxy. Now this somehow means jibri doesn't see the `meet.jitsi` network? But it's included in the network.
Would it not be easier to share a volume between `web` and `jibri` and use that generated .crt directly?

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert (hunk 2 above)
Comment (albe) [inline jibri/rootfs/etc/cont-init.d/11-get-https-cert:6]: why is the screenshot to create the `~/.pki` required? I could not find any trace of this file being required on the net

Comment (teowoz) [reply jibri/rootfs/etc/cont-init.d/11-get-https-cert:6]: - while loop - retries obtaining the certificate in case the web container hasn't started yet.
- openssl inside while loop - the first openssl connects to web server and returns certificate on stdout, the second one saves the certificate in PEM format
- update-ca-certificates - redundant, can be removed. Chrome doesn't use system-wide certificate storage.
- su jibri -s /bin/sh -c - runs the quoted command as user jibri, to prevent permissions problems
- google-chrome --headless --screenshot ... - as explained in the comment
- certutil -d sql:/home/jibri/.pki/nssdb -A ... - adds certificate we've just obtained to the Chrome database

File: web/rootfs/defaults/ssl.conf (hunk 3 above)
Comment (teowoz) [reply web/rootfs/defaults/ssl.conf:25]: Not really needed, changed it while experimenting and overlooked it.

File: web/rootfs/etc/cont-init.d/10-config (hunk 4 above)
Comment (teowoz) [reply web/rootfs/etc/cont-init.d/10-config:46]: Why not? The certificate is just to satisfy the browser, not to provide real security.

File: web/rootfs/defaults/ssl.conf (hunk 3 above)
[the remark this answers is not in the document]
Comment (teowoz) [reply web/rootfs/defaults/ssl.conf:25]: No. I changed it during experiments and overlooked it.

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert (hunk 1 above)
Comment (teowoz) [reply jibri/rootfs/etc/cont-init.d/11-get-https-cert:3]: If you've purged the config dir, maybe the web container was generating dhparams file, which can take a while before nginx starts.

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert (hunk 2 above)
Comment (teowoz) [reply jibri/rootfs/etc/cont-init.d/11-get-https-cert:6]: I haven't checked whether Chrome creates the database as empty. If so, your version would be simpler and faster.

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert (hunk 1 above)
Comment (teowoz) [reply jibri/rootfs/etc/cont-init.d/11-get-https-cert:3]: Shared volume would be equally good, I think. In both cases you need the retrying loop (or generating the certificates before running docker-compose). In both cases you need to add the certificate to the database. So it would be slightly simpler (cp instead of openssl), but would need additional Docker volume.

**Comment by teowoz** [comment]
From what I see in #240 discussions, this PR needs testing on completely fresh Jitsi Meet instance.

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert (hunk 1 above)
Comment (albe) [reply jibri/rootfs/etc/cont-init.d/11-get-https-cert:3]: Slightly changed that part and manually transfered the certificate now:
```suggestion
while [ ! -f /usr/local/share/ca-certificates/meet.crt ]
do
  sleep 1
  if CERTS=$(openssl s_client -showcerts -connect meet.jitsi:443 < /dev/null); then
    echo "$CERTS" | openssl x509 -outform PEM > /usr/local/share/ca-certificates/meet.crt
  fi
done
```
Result is this:
```
jibri_1     | Updating certificates in /etc/ssl/certs...
jibri_1     | 1 added, 0 removed; done.
jibri_1     | Running hooks in /etc/ca-certificates/update.d...
jibri_1     |
jibri_1     | Adding debian:meet.pem
jibri_1     | done.
jibri_1     | done.
jibri_1     | Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
jibri_1     | [0410/173754.459884:INFO:headless_shell.cc(619)] Written to file /tmp/s.png.
jibri_1     | certutil: function failed: SEC_ERROR_BAD_DATABASE: security library: bad database.
jibri_1     | [cont-init.d] 11-get-https-cert: exited 0.
jibri_1     | [cont-init.d] done.
```
:/
After initializing the database with certutil like below this error doesn't come up, but I'm still having issues with connecting to `meet.jitsi:443` from the jibri container. I doubt that is related to this fix, but maybe also a root cause for the issue it (I) tries to solve.

File: web/rootfs/etc/cont-init.d/10-config (hunk 4 above)
Comment (saghul) [reply web/rootfs/etc/cont-init.d/10-config:46]: I know, just wondering if the change was deliberate or there was something I was missing.

**Comment by macoronado** [comment]
> 
> 
> @ricardona
> 
> Did it successfully add the certificate?
> It should print:
> 
> > Updating certificates in /etc/ssl/certs...
> > 1 added, 0 removed; done.
> 
> during start.
> Before checking it, remove (stopping is not enough) the jibri container.

I have the same problem as @ricardona. The certificate is added. After that I got an SSL error with the screeshot but i don't know if its related to the timeout:

> depth=0 CN = meet.jitsi
> verify error:num=18:self signed certificate
> verify return:1
> depth=0 CN = meet.jitsi
> verify return:1
> DONE
> Updating certificates in /etc/ssl/certs...
> 1 added, 0 removed; done.
> Running hooks in /etc/ca-certificates/update.d...
> done.
> done.
> Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
> [0415/122429.837949:ERROR:command_buffer_proxy_impl.cc(125)] ContextResult::kTransientFailure: Failed to send GpuChannelMsg_CreateCommandBuffer.
> [0415/122430.114002:ERROR:cert_verify_proc_nss.cc(998)] CERT_PKIXVerifyCert for meet.jitsi failed err=-8179
> [0415/122430.114675:ERROR:ssl_client_socket_impl.cc(941)] handshake failed; returned -1, SSL error code 1, net_error -202
> [0415/122430.257603:INFO:headless_shell.cc(619)] Written to file /tmp/s.png.
> [cont-init.d] 11-get-https-cert: exited 0.
> [cont-init.d] done.

Also, I have seen that in browser.0.txt there is a recurrent error until the timeout happens:

> Cannot read property 'isJoined' of undefined

[browser.0.txt](https://github.com/jitsi/docker-jitsi-meet/files/4480771/browser.0.txt)

I have seen in the forums that some people also got that error with JWT enabled (https://community.jitsi.org/t/solved-jibri-with-jwt-token-enabled-jitsi/16646/29), but in the docker image "c2s_require_encryption" is set to false by default.

File: web/rootfs/defaults/ssl.conf (hunk 3 above)
Comment (teowoz) [reply web/rootfs/defaults/ssl.conf]: Restored it. It makes difference for Jitsi Meet exposed to the World without reverse proxy.

[the remark this answers is not in the document]
Comment (teowoz) [reply web/rootfs/defaults/ssl.conf]: Restored it. It makes difference for Jitsi Meet exposed to the World without reverse proxy.

**Comment by teowoz** [comment]
@ricardona 
> @teowoz I tried [your instructions](https://github.com/jitsi/docker-jitsi-meet/issues/240#issuecomment-611633063) but I got:
> 
> ```
> jibri_1    | 2020-04-10 00:25:15.393 FINE: [61] org.jitsi.jibri.selenium.pageobjects.CallPage.visit()Visiting url https://meet.jitsi/pimpollo#config.iAmRecorder=true&config.externalConnectUrl=null&config.startWithAudioMuted=true&config.startWithVideoMuted=true&interfaceConfig.APP_NAME="Jibri"&config.analytics.disabled=true&config.p2p.enabled=false
> 
> jibri_1    | 2020-04-10 00:25:46.112 SEVERE: [61] org.jitsi.jibri.selenium.pageobjects.CallPage.visit() Timed out waiting for call page to load
> ```

I've also had this problem recently, but not on my previous install.
After inspecting browser logs I saw CORS errors.
You'll need to allow CORS requests from `meet.jitsi` in world-facing https server. For example in Apache:

```
SetEnvIf Origin "^http(s)?://(www\.)?(meet\.jitsi|your-jitsi-meet-domain\.example\.com)$" AccessControlAllowOrigin=$0
Header add Access-Control-Allow-Origin %{AccessControlAllowOrigin}e env=AccessControlAllowOrigin
Header set Access-Control-Allow-Headers Content-Type
Header add Vary Origin
```

**Comment by miguelpeixe** [comment]
From my understanding, this current proposal still doesn't fix the problem for live instances, because it does not consider adding a nginx entry for internal meet.jitsi fqdn with the self-signed cert for builds that uses letsencrypt certs.

Should'nt this certificate always be generated by the web image and added to nginx as an internal entry?

File: jibri/rootfs/etc/cont-init.d/11-get-https-cert [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/with-contenv bash
+
+while [ ! -f /usr/local/share/ca-certificates/meet.crt ]
+do
+  sleep 1
+  if CERTS=$(openssl s_client -showcerts -connect meet.jitsi:443 < /dev/null); then
```
Comment (danoh) [inline jibri/rootfs/etc/cont-init.d/11-get-https-cert:6]: the openssl works for me only with SNI like this
`openssl s_client -showcerts -connect meet.jitsi:443 -servername meet.jitsi`

**Comment by saghul** [comment]
Solved in https://github.com/jitsi/docker-jitsi-meet/pull/1046 with a different approach: Jibri will not be opening the real public URL.
